src/server/src/index.ts

```diff
@@ -26,74 +49,128 @@ app.get("/", (req, res) => {
   });
 });
 
-let config = { width: 16, height: 16, mines: 40 };
-let game = new GameState(config);
+// Endpoint to check if a lobby exists
+app.post("/check-lobbies", (req, res) => {
+  const { lobby } = req.body;
+  const isInSet = lobbies.has(lobby);
+  res.json({ isInSet });
+});
+
+// Endpoint to create a new game lobby
+app.post("/create-lobby", (req, res) => {
+  const { gameConfig, room } = req.body;
+  console.log(req.body);
+  console.log(`Creating Room ${room}`);
 
-const connections: Dictionary<Socket> = {};
-const users: Dictionary<User> = {};
+  // Initialize game state for the new room
+  gameStore[room] = {
+    board: new GameState(gameConfig),
+    config: gameConfig,
+    users: {},
+    connections: {},
+  };
+  lobbies.add(room);
+  res.send({
+    status: "ok",
+    message: "Minesweeper lobby generated",
+    timestamp: new Date().toISOString(),
+  });
+});
 
-const handleMovement = (cursorPosition: User["state"], uuid: string) => {
-  users[uuid] = {
-    ...users[uuid],
+// Handler for cursor movement updates
+const handleMovement = (
+  cursorPosition: User["state"],
+  uuid: string,
+  room: string
+) => {
+  // Update user's cursor position in game state
+  gameStore[room].users[uuid] = {
+    ...gameStore[room].users[uuid],
     state: cursorPosition,
   };
-  io.emit("users", users);
+
+  // Broadcast updated user positions to all clients in room
+  io.to(room).emit("users", gameStore[room].users);
 };
 
-const handleClose = (uuid: string) => {
-  console.log(`Disconnecting ${users[uuid].username}`);
-  delete connections[uuid];
-  delete users[uuid];
-  console.log(users);
-  io.emit("users", users);
+// Handler for closed connection
+const handleClose = (uuid: string, room: string) => {
+  console.log(`Disconnecting ${gameStore[room].users[uuid].uuid}`);
+  delete gameStore[room].connections[uuid];
+  delete gameStore[room].users[uuid];
+  console.log(gameStore[room].users);
+  io.to(room).emit("users", gameStore[room].users);
 };
 
+// Socket.IO connection handler
 io.on("connection", (socket) => {
+  // Extract username from connection query
   const username = String(socket.handshake.query["username"]);
+  // Extract room ID from connection query
+  const room = String(socket.handshake.query["room"]);
+  // Generate unique ID for new user
   const uuid: string = uuidv4();
-  console.log(`${username} connected with uuid ${uuid}`);
-  connections[uuid] = socket;
-  users[uuid] = {
+
+  socket.join(room);
+
+  if (!gameStore[room]) {
```

Comment:
Not technically necessary since to be able to join a game, a room MUST be created, but nice to have nonetheless